pkg/controller/tenant/tenant_controller.go

```diff
@@ -278,33 +300,61 @@ func (tc *TenantController) createNamespaces(tenant string) (error, bool) {
 }
 
 func (tc *TenantController) createInitialRoleAndBinding(tenant string) (error, bool) {
-	// tenant admin act as cluster admin for tha tenant
-	role := &rbacv1.ClusterRole{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   InitialClusterRoleName,
-			Tenant: tenant,
-		},
-		Rules: initialClusterRoleRules(),
+
```

Comment:
Same: Need to check the tenant's status is active before creating. Otherwise we might run into weird race condition issue when deleting a tenant.